src/Common/Libraries/Uplink/Controller.php

```diff
@@ -0,0 +1,174 @@
+<?php
```

Comment:
If all is required by the  Uplink functionality is one file, why not move it to the root?
From `./Uplink/Controller.php` to `./Uplink.php`.